src/services_models/services_constructor.jl

```diff
@@ -10,31 +10,42 @@ function construct_services!(psi_container::PSIContainer,
     isempty(services_template) && return
     services_mapping = PSY.get_contributing_device_mapping(sys)
     for service_model in values(services_template)
-        contributing_devices = _filter_service_mapping(service_model.service_type, services_mapping)
-        for value in values(contributing_devices)
-            add_service!(psi_container,
-                         value.service,
-                         service_model,
-                         value.contributing_devices,
-                         devices_template; kwargs...)
-        end
+        service_devices = _filter_service_mapping(service_model.service_type, services_mapping)
+        services = PSY.get_components(service_model.service_type, sys)
+        construct_service!(psi_container,
+                            services,
+                            service_devices,
+                            service_model,
+                            devices_template; kwargs...)
     end
     return
 end
 
-function add_service!(psi_container::PSIContainer, service::SR,
+function construct_service!(psi_container::PSIContainer, 
+                      services::IS.FlattenIteratorWrapper{SR},
+                      service_devices::Dict{NamedTuple{(:type, :name),Tuple{DataType,String}},PSY.ServiceContributingDevices},
```

Comment:
We need to pass here the contributing devices and perform the parsing inside. This object is not easy to track.